pnpm.fetchDeps: error if lockfile version has a mismatch (#317739)
NixOS/nixpkgs: doc/languages-frameworks/javascript.section.md

})
```

> It is highly recommended to use a pinned version of pnpm (i.e. `pnpm_8` or `pnpm_9`), to increase future reproducibility. It might also be required to use an older version, if the package needs support for a certain lock file version. The fetchDeps hook will abort, if the used pnpm version is to old.

Reviewer: Typo
```suggestion
It is highly recommended to use a pinned version of pnpm (i.e. `pnpm_8` or `pnpm_9`), to increase future reproducibility. It might also be required to use an older version, if the package needs support for a certain lock file version. The fetchDeps hook will abort if the used pnpm version is too old.
```

> With `removePackageManagerVersion` that version requirement can be relaxed if desired.

Reviewer: ```suggestion
With `removePackageManagerVersion`, that version requirement can be relaxed if desired.
```

> It is highly recommended to use a pinned version of pnpm (i.e. `pnpm_8` or `pnpm_9`), to increase future reproducibility. It might also be required to use an older version, if the package needs support for a certain lock file version. The fetchDeps hook will abort if the used pnpm version is too old.
> If desired, that version requirement can be relaxed via `removePackageManagerVersion`.

Reviewer: The phrasing is still honestly a bit odd. Also, it might be better to rename the parameter to something like `relaxPnpmVersion` 

```suggestion
It is highly recommended to use a pinned version of pnpm (i.e. `pnpm_8` or `pnpm_9`), to increase future reproducibility. It might also be required to use an older version, if the package needs support for a certain lock file version.
Unless the `relaxPnpmVersion` parameter is set to true, the fetchDeps hook will abort if the used pnpm version is older than the `lockfileVersion` specified in the lockfile.
```
